aci/data_source_aci_cloudtemplateextnetwork.go

```diff
@@ -0,0 +1,109 @@
+package aci
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	"github.com/ciscoecosystem/aci-go-client/v2/client"
+	"github.com/ciscoecosystem/aci-go-client/v2/models"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+func dataSourceAciCloudTemplateforExternalNetwork() *schema.Resource {
+	return &schema.Resource{
+		ReadContext:   dataSourceAciCloudTemplateforExternalNetworkRead,
+		SchemaVersion: 1,
+		Schema: AppendBaseAttrSchema(AppendNameAliasAttrSchema(map[string]*schema.Schema{
+			"hub_network_name": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"vrf_dn": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"regions": {
+				Type: schema.TypeList,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+				Optional: true,
+				Computed: true,
+			},
+			"cloud_vendor": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ValidateFunc: validation.StringInSlice([]string{
```

Comment:
do we need validatefunc in the dataSource?

Reply:
Yes. Since the user cannot input other values.